src/game/types/guards/index.ts

```diff
@@ -0,0 +1,63 @@
+import { CardType, ICrop, IField, IGame, IPlayer, ITable } from '../'
+
+export const isCrop = (obj: unknown): obj is ICrop => {
+  if (typeof obj !== 'object' || obj === null) return false
+
+  return (
+    'id' in obj &&
+    typeof obj.id === 'string' &&
+    'type' in obj &&
+    obj.type === CardType.CROP
```

Comment:
some of this feels a little redundant. `typeof obj.id` will not return "string" if id doesn't exist, so i'm wondering what the point of the "in" checks is (this comment applies to all the methods in this file it looks like)

Reply:
ahh gotcha. ok that makes sense. i wasn't thinking about typescript quirks. thanks for the explanation!